kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/KeyValueHeaderRecordWriterProvider.java

```diff
@@ -95,7 +95,8 @@ public void write(SinkRecord sinkRecord) {
         }
 
         // headerWriter != null means writing headers is turned on
-        if (headerWriter != null && sinkRecord.headers() == null) {
+        if (headerWriter != null
```

Comment:
@dosvath I find the combination of `if` and then `if else` more confusing than it's required. 

I'd suggest, with the opportunity of this bugfix to change the branches above and here as: 

```
        if (keyWriter != null) {
          if (sinkRecord.key() == null) {
            throw new DataException(
                String.format("Key cannot be null for SinkRecord: %s", sinkRecord)
            );
          }
          keyWriter.write(sinkRecord);
        }
```
and then
```
        if (headerWriter != null) {
          if (sinkRecord.headers() == null || sinkRecord.headers().isEmpty()) {
            throw new DataException(
                String.format("Headers cannot be null for SinkRecord: %s", sinkRecord)
            );
          }
          headerWriter.write(sinkRecord);
        }
```

Reply:
I agree 👍